types/jest/index.d.ts

```diff
@@ -976,28 +976,37 @@ declare namespace jest {
     }
 
     /**
-     * Represents the result of a single call to a mock function.
+     * Represents the result of a single call to a mock function with a return value.
      */
-    interface MockResult {
-        /**
-         * True if the function threw.
-         * False if the function returned.
-         */
-        isThrow: boolean;
-        /**
-         * The value that was either thrown or returned by the function.
-         */
+    interface MockResultReturn<T> {
+        type: 'return';
+        value: T;
+    }
+    /**
+     * Represents the result of a single incomplete call to a mock function.
+     */
+    interface MockResultIncomplete {
+        type: 'incomplete';
+        value: undefined;
+    }
+    /**
+     * Represents the result of a single call to a mock function with a thrown error.
+     */
+    interface MockResultThrow {
```

Comment:
This should be generic too, but then it will be harder to define MockResult type

Reply:
There is no way of inferring exceptions from a function in Typescript. https://github.com/Microsoft/TypeScript/issues/13219

Since exceptions can't be inferred, we have to use any type here.

Reply:
Ok !